src/main/java/net/dv8tion/jda/internal/requests/ratelimit/BotRateLimiter.java

```diff
@@ -102,6 +103,33 @@ private ScheduledExecutorService getScheduler()
         return requester.getJDA().getRateLimitPool();
     }
 
+    @Override
+    public int cancelRequests()
+    {
+        return MiscUtil.locked(bucketLock, () -> {
+            for (Future<?> future : rateLimitQueue.values())
+                future.cancel(false);
+            rateLimitQueue.clear();
+
+            AtomicInteger count = new AtomicInteger(0);
+            buckets.values()
+                .stream()
+                .map(Bucket::getRequests)
+                .flatMap(Collection::stream)
+                .filter(request -> !request.isPriority())
+                .forEach(request -> {
+                    request.cancel();
+                    count.incrementAndGet();
+                });
+            buckets.clear();
+            if (count.get() == 1)
+                RateLimiter.log.warn("Cancelled 1 request!");
+            else if (count.get() > 1)
+                RateLimiter.log.warn("Cancelled {} requests!", count.get());
```

Comment:
Why do we bother differentiating between 1 and > 1?
Also, do we really want to make clearing a warn? Seems like it could be an annoying message to have popup. Curious as to the thought process behind the logging level.

Reply:
We also call `count.get()` 4 times here. Can we relegate the result to a variable instead?

Reply:
It's a warn because it should be the last resort. Doing this regularly in normal operation should be considered a bug. I differentiate because why not.

Reply:
Ah. you differentiate for the `s`.